src/include/migraphx/op/multibroadcast.hpp

```diff
@@ -102,19 +102,19 @@ struct multibroadcast
         }
         else
         {
-            // two inputs
-            auto s1 = inputs.at(1);
-            if(s0.dynamic() or s1.dynamic())
+            // 2+ inputs
+            if(std::any_of(
+                   inputs.cbegin(), inputs.cend(), [](auto input) { return input.dynamic(); }))
             {
                 if(not output_dyn_dims.empty())
                 {
                     return {t, output_dyn_dims};
                 }
-                return {t, compute_broadcasted_dyn_dims(s0, s1)};
+                return {t, compute_common_dyn_dims(inputs)};
             }
             else
             {
-                auto bcast_lens    = compute_broadcasted_lens(s0.lens(), s1.lens());
+                auto bcast_lens    = compute_common_lens(inputs);
```

Comment:
Could we do similar check like dynamic shape here that if `out_lens` is set then, no need to recalculate common_lens again ?

Reply:
No, `out_lens` will never be set for 2+ input version of multibroadcast. It is only used for the 1 input version. If 2+ input multibroadcast uses only static shapes it should be converted to a one input version. We might have to make that a compiler pass in the future.

Reply:
It would be better to put that into comments somewhere that `out_lens` is never set for static shape 2+ inputs. 
But `dyn_out_lens` is set for 2+ input case because during runtime, `dyn_out_lens` wouldn't be used and 2+ inputs would have been converted into static shapes.